expat/tests/acc_tests.c

```diff
@@ -54,180 +54,191 @@
-          "<?xml version='1.0' encoding='utf-8'?>\n"
-          "hello\n"
-          "xml\n" /* with trailing newline! */,
-          0,
-      },
-      {"<!DOCTYPE doc SYSTEM 'first.ent'>\n"
-       "<doc></doc>\n",
-       "<!ELEMENT doc EMPTY>\n"
-       "<!ENTITY % e1 SYSTEM 'second.ent'>\n"
-       "<!ENTITY % e2 '%e1;'>\n"
-       "%e1;\n",
-       "\xEF\xBB\xBF<!ATTLIST doc a1 CDATA 'value'>" /* UTF-8 BOM */,
-       strlen("\xEF\xBB\xBF<!ATTLIST doc a1 CDATA 'value'>")},
+    {"<e/>", NULL, NULL, 0},
+    {"<e></e>", NULL, NULL, 0},
+
+    /* Attributes */
+    {"<e k1=\"v2\" k2=\"v2\"/>", NULL, NULL, 0},
+    {"<e k1=\"v2\" k2=\"v2\"></e>", NULL, NULL, 0},
+    {"<p:e xmlns:p=\"https://domain.invalid/\" />", NULL, NULL, 0},
+    {"<e k=\"&amp;&apos;&gt;&lt;&quot;\" />", NULL, NULL,
+     sizeof(XML_Char) * 5 /* number of predefined entities */},
+    {"<e1 xmlns='https://example.org/'>\n"
+     "  <e2 xmlns=''/>\n"
+     "</e1>",
+     NULL, NULL, 0},
+
+    /* Text */
+    {"<e>text</e>", NULL, NULL, 0},
+    {"<e1><e2>text1<e3/>text2</e2></e1>", NULL, NULL, 0},
+    {"<e>&amp;&apos;&gt;&lt;&quot;</e>", NULL, NULL,
+     sizeof(XML_Char) * 5 /* number of predefined entities */},
+    {"<e>&#65;&#41;</e>", NULL, NULL, 0},
+
+    /* Prolog */
+    {"<?xml version=\"1.0\"?><root/>", NULL, NULL, 0},
+
+    /* Whitespace */
+    {"  <e1>  <e2>  </e2>  </e1>  ", NULL, NULL, 0},
+    {"<e1  ><e2  /></e1  >", NULL, NULL, 0},
+    {"<e1><e2 k = \"v\"/><e3 k = 'v'/></e1>", NULL, NULL, 0},
+
+    /* Comments */
+    {"<!-- Comment --><e><!-- Comment --></e>", NULL, NULL, 0},
+
+    /* Processing instructions */
+    {"<?xml-stylesheet type=\"text/xsl\" href=\"https://domain.invalid/\" media=\"all\"?><e/>",
+     NULL, NULL, 0},
+    {"<?pi0?><?pi1 ?><?pi2  ?><r/><?pi4?>", NULL, NULL, 0},
+#  ifdef XML_DTD
+    {"<?pi0?><?pi1 ?><?pi2  ?><!DOCTYPE r SYSTEM 'first.ent'><r/>",
+     "<?pi3?><!ENTITY % e1 SYSTEM 'second.ent'><?pi4?>%e1;<?pi5?>", "<?pi6?>",
+     0},
+#  endif /* XML_DTD */
+
+    /* CDATA */
+    {"<e><![CDATA[one two three]]></e>", NULL, NULL, 0},
+  /* The following is the essence of this OSS-Fuzz finding:
+     https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=34302
+     https://oss-fuzz.com/testcase-detail/4860575394955264
+  */
+#  if XML_GE == 1
+    {"<!DOCTYPE r [\n"
+     "<!ENTITY e \"111<![CDATA[2 <= 2]]>333\">\n"
+     "]>\n"
+     "<r>&e;</r>\n",
+     NULL, NULL, sizeof(XML_Char) * strlen("111<![CDATA[2 <= 2]]>333")},
+#  endif /* XML_GE == 1 */
+
+#  ifdef XML_DTD
+    /* Conditional sections */
+    {"<!DOCTYPE r [\n"
+     "<!ENTITY % draft 'INCLUDE'>\n"
+     "<!ENTITY % final 'IGNORE'>\n"
+     "<!ENTITY % import SYSTEM \"first.ent\">\n"
+     "%import;\n"
+     "]>\n"
+     "<r/>\n",
+     "<![%draft;[<!--1-->]]>\n"
+     "<![%final;[<!--22-->]]>",
+     NULL, sizeof(XML_Char) * (strlen("INCLUDE") + strlen("IGNORE"))},
+#  endif /* XML_DTD */
+
+  /* General entities */
+#  if XML_GE == 1
```

Comment:
Redundant check (whole function is already covered).

Reply:
Confirming, will be gone with the next push in a few minutes. Thanks!